### PR TITLE
Lambda to tag EC2 instances & ECS cluster and container instance id

### PR DESCRIPTION
**Pull request by kenoir**
### What is this PR trying to achieve?

Simpler identification of EC2 instance participation in an ECS cluster.

### Who is this change for?

Developers who want simpler logic when mapping EC2 instances to ECS containers

### Have the following been considered/are they needed?

- [ ] Run `terraform apply`.

### CODE REVIEW

File: lambdas/ecs_ec2_instance_tagger/ecs_ec2_instance_tagger.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+{
+  "source": [
+    "aws.ecs"
+  ],
+  "detail-type": [
+    "ECS Container Instance State Change"
+  ]
+}
+
```
Comment (alexwlchan) [inline lambdas/ecs_ec2_instance_tagger/ecs_ec2_instance_tagger.py]: 👍 for the nice description, especially the event pattern.

Some of the AWS terminology makes this very wordy and hard to follow though (not your fault!). Could we break it up a little? Maybe:

> Tag EC2 instances with their cluster ARN and container instance ARN. This allows us to easily follow an instance back to its ECS cluster.
>
> This script is triggered by an ECS Container Instance State Change event.

File: lambdas/ecs_ec2_instance_tagger/ecs_ec2_instance_tagger.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+"""
+
+import pprint
+
+import boto3
+
+
+def create_tags(ec2_client, ec2_instance_id, event_detail):
+    """
+    Given an ECS cluster, service name and desired instance count, change
```
Comment (alexwlchan) [inline lambdas/ecs_ec2_instance_tagger/ecs_ec2_instance_tagger.py]: This docstring is copied from a different function?

File: lambdas/ecs_ec2_instance_tagger/ecs_ec2_instance_tagger.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+def main(event, _):
+    pprint.pprint(event)
+    ec2_client = boto3.client('ec2')
+
+    ec2_instance_id = event['detail']['ec2InstanceId']
+
+    response = create_tags(ec2_client, ec2_instance_id, event['detail'])
+
+    pprint.pprint(response)
```
Comment (alexwlchan) [inline lambdas/ecs_ec2_instance_tagger/ecs_ec2_instance_tagger.py]: nit: needs a trailing newline

File: lambdas/ecs_ec2_instance_tagger/ecs_ec2_instance_tagger.py (hunk 1 above)
Comment (kenoir) [inline lambdas/ecs_ec2_instance_tagger/ecs_ec2_instance_tagger.py]: 👍
